packages/rocketchat-custom-oauth/custom_oauth_server.coffee

```diff
@@ -135,7 +144,7 @@ class CustomOAuth
 				serviceData: serviceData
 				options:
 					profile:
-						name: identity.name or identity.username or identity.nickname or identity.CharacterName
+						name: identity.name or identity.username or identity.nickname or identity.CharacterName or identity.user.name
```

Comment:
Can you protect the code replacing `identity.user.name` by `identity.user?.name`